src/GenericController.js

```diff
@@ -0,0 +1,17 @@
+export default class GenericController {
+  constructor() {
+    this.controls = {};
+    const that = this;
+    this.addMethod('init', function init() {
+      that.emitter = this;
+    });
+  }
+
+  addMethod(methodName, method) {
+    this.controls[methodName] = method;
+  }
+
+  getController() {
```

Comment:
Instead of `getController` which isn't intuitive (I think I would get the same object since this is already a controller), maybe name this `toObject()`?